### ISSUE TITLE
Inconsistent/confusing casing of AWS IAM OpenIdConnectProvider

### ISSUE THREAD

**Issue by eydward** [pulumi/docs#9452]
In the AWS classic provider we appear to have inconsistent capitalization, perhaps caused by variance between languages. It would be nice to make this appear more consistent (or at least clear up which names are which on the docs).

c.f.
- [aws.iam.OpenIdConnectProvider](https://www.pulumi.com/registry/packages/aws/api-docs/iam/openidconnectprovider/)
- and [aws.iam.getOpenidConnectProvider](https://www.pulumi.com/registry/packages/aws/api-docs/iam/getopenidconnectprovider/#aws-iam-getopenidconnectprovider) has different casing
- and in the example block of above, `aws.iam.get_openid_connect_provider`

**Comment by cnunciato** [comment]
@eydward I'm not sure I totally understand what's being called out here, as it sounds like it could be a few things:

1. The difference in casing of "id" between `OpenIdConnectProvider` in the first example (PascalCase) and `OpenidConnectProvider` in the second (lowercase)
2. The difference in casing between `Open...` (PascalCase) and `getOpen...` (camelCase)
3. That Python is completely different (snake_case)
4. That the titles of the pages are one thing (our.simplified.Types), but the Python example is something else. 

The first three are really outside the realm of docs, as they're language-based and derived from the schemas themselves. I don't think there's much we can do about these, unfortunately.

The fourth, though, while "by design" as of today, does make things a bit harder to find if you're using something like Python and looking specifically for `aws.get_openid_connect_provider`. Searching for that particular string does show the right result, but because the result doesn't look snake-cased, it may not be clear it's the thing you're looking for. 

Would you mind adding a bit more detail about what it is this is about, and maybe what you think a good solution might look like?

**Comment by eydward** [comment]
Yes - I now realize that most of this is actually not a docs issue. I do still wonder re the fourth point if there is a way to not confuse our Python friends who are looking for get_open_id_connect_provider with a title page of GetOpenIdConnectProvider, but I think this issue can probably just be closed/deleted.

**Comment by scottslowe** [comment]
@interurban Assigning to you, probably needs to go to the Docs board

**Comment by cnunciato** [comment]
We've talked through this one and decided to close for now, as the naming we have today is still pretty much by design.
